doc/source/conf.py

```diff
@@ -174,23 +183,21 @@ def generate_rst_files(versions: list[str], tables: dict[str, list[str]]):
         output_filename = GENERATED_DIR / f"version_{version}.rst"
 
         # Write the rendered content to the file
-        with open(output_filename, "w") as f:
-            f.write(rendered_content)
+        with Path.open(output_filename, "w") as file:
+            file.write(rendered_content)
```

Comment:
For this operation, you can simply do:

```suggestion
        output_filename.write_text(rendered_content, encoding="utf-8")
```

Please, apply this in other situations within this file.

Reply:
Thanks for the feedback Jorge. Flimsy oversight on my part, ruff recommended to use Path.open() and once I saw it could be called as a class method, I never looked back.

Can I still do the writing within the open context manager? Since it is designed to ensure that a file will be closed even in the event that the code raises an error.
@jorgepiloto